src/Apps/Order/Routes/Shipping2/Hooks/useSaveSelectedShippingQuote.tsx

```diff
@@ -36,12 +36,21 @@ export const useSaveSelectedShippingQuote = (
     try {
       shippingContext.actions.setIsPerformingOperation(true)
 
+      if (
+        shippingContext.state.fulfillmentDetailsCtx?.values.meta.mode ===
+        "new_address"
+      ) {
+        await handleNewUserAddressUpdates(
+          shippingContext.state.fulfillmentDetailsCtx.values
```

Comment:
An effect of letting a user edit values in 2 steps at once - if they _change_ the address, the shipping quotes will disappear and they will be back on the fulfillment details step. But if they check/uncheck the box we just need to save/delete the address - hence this line, which for consistency now happens before saving the shipping quote.

Another way to do this could be to have the save/unsave mutation triggered from inside fulfillment details immediately upon clicking the box. Might be a good follow-up.